app/actions/remote/thread.ts

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {switchToThread} from '@actions/local/thread';
+import {fetchPostThread} from '@actions/remote/post';
+
+export const fetchAndSwitchToThread = async (serverUrl: string, rootId: string) => {
+    // Load thread before we open to the thread modal
+    fetchPostThread(serverUrl, rootId);
```

Comment:
Most of the time, we will have the `reply` and `root post` before we go to the screen. The only problem with `await` is, it will delay the opening of the Modal, and the experience might not be good.